ddl/table.go

```diff
@@ -170,6 +170,28 @@ func (d *ddl) getTable(t *meta.Meta, schemaID int64, tblInfo *model.TableInfo) (
 	return tbl, nil
 }
 
+func (d *ddl) getTableInfo(t *meta.Meta, job *model.Job) (*model.TableInfo, error) {
+	schemaID := job.SchemaID
+	tableID := job.TableID
+	tblInfo, err := t.GetTable(schemaID, tableID)
+	if errors2.ErrorEqual(err, meta.ErrDBNotExists) {
+		job.State = model.JobCancelled
+		return nil, errors.Trace(ErrNotExists)
+	} else if err != nil {
```

Comment:
It is the old code i moved from index.go.
How about u? @siddontang 
Seems it is better to remove `job.State = model.JobCancelled` out of getTableInfo function.